lib/AST/ASTContext.cpp

```diff
@@ -5166,45 +5187,66 @@ CanGenericSignature ASTContext::getSingleGenericParameterSignature() const {
   return canonicalSig;
 }
 
-// Return the signature for an opened existential. The opened archetype may have
-// a different set of conformances from the corresponding existential. The
-// opened archetype conformances are dictated by the ABI for generic arguments,
-// while the existential value conformances are dictated by their layout (see
-// Type::getExistentialLayout()). In particular, the opened archetype signature
-// does not have requirements for conformances inherited from superclass
-// constraints while existential values do.
-CanGenericSignature ASTContext::getOpenedArchetypeSignature(Type type) {
+Type OpenedArchetypeType::getSelfInterfaceTypeFromContext(const DeclContext *useDC) {
+  auto typeContext = useDC->getInnermostTypeContext();
+  if (typeContext && typeContext->getSelfProtocolDecl()) {
+    return typeContext->getProtocolSelfType();
+  } else {
+    return GenericTypeParamType::get(
+        /*isTypeSequence=*/false,
+        /*depth*/ useDC->getGenericContextDepth() + 1, /*index*/ 0,
```

Comment:
Again, this should take the outer GenericSignature directly instead of a DeclContext.